manager/manager.go

```diff
@@ -995,21 +1034,23 @@ func (self *manager) watchForNewContainers(quit chan error) error {
 	go func() {
 		for {
 			select {
-			case event := <-eventsChannel:
+			case event := <-self.eventsChannel:
 				switch {
-				case event.EventType == container.SubcontainerAdd:
-					err = self.createContainer(event.Name)
-				case event.EventType == container.SubcontainerDelete:
+				case event.EventType == watcher.SubcontainerAdd:
+					err = self.createContainer(event.Name, event.WatchSource)
+				case event.EventType == watcher.SubcontainerDelete:
 					err = self.destroyContainer(event.Name)
+				case event.EventType == watcher.SubcontainerOverride:
+					err = self.overrideContainer(event.Name, event.WatchSource)
 				}
 				if err != nil {
 					glog.Warningf("Failed to process watch event %+v: %v", event, err)
 				}
 			case <-quit:
 				// Stop processing events if asked to quit.
-				err := root.handler.StopWatchingSubcontainers()
-				quit <- err
-				if err == nil {
+				for _, watcher := range self.containerWatchers {
+					err := watcher.StopWatchingSubcontainers()
+					quit <- err
```

Comment:
this is the exact same code as before.  We spoke before about this being sort of pointless, perhaps its ok to not return an error and just signal by closing the quit channel?